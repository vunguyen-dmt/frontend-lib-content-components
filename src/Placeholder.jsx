import React from 'react';

const Placeholder = () => (
  <div className="Placeholder">
    <h1>
      Under Construction
      <br />
      Coming Soon
    </h1>
  </div>
);

export default Placeholder;
