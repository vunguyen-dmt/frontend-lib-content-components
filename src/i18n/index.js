import arMessages from './messages/ar.json';
// no need to import en messages-- they are in the defaultMessage field
import es419Messages from './messages/es_419.json';
import frMessages from './messages/fr.json';
import kokrMessages from './messages/ko_KR.json';
import ptbrMessages from './messages/pt_BR.json';
import zhcnMessages from './messages/zh_CN.json';
import viMessages from './messages/vi.json';

const messages = {
  ar: arMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  'ko-kr': kokrMessages,
  'pt-br': ptbrMessages,
  vi: viMessages,
};

export default messages;
