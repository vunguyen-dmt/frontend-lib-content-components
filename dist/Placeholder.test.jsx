import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import TestRenderer from 'react-test-renderer';
import { AppContext } from '@edx/frontend-platform/react';
import { Context as ResponsiveContext } from 'react-responsive';

import Placeholder from './index';

describe('<Placeholder />', () => {
  it('renders correctly', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1280 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider
            value={{
              authenticatedUser: null,
              config: {
                LMS_BASE_URL: process.env.LMS_BASE_URL,
                SITE_NAME: process.env.SITE_NAME,
                LOGIN_URL: process.env.LOGIN_URL,
                LOGOUT_URL: process.env.LOGOUT_URL,
                LOGO_URL: process.env.LOGO_URL,
              },
            }}
          >
            <Placeholder />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
