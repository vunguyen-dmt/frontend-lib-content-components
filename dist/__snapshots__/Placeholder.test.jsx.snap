// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Placeholder /> renders correctly 1`] = `
<div
  className="Placeholder"
>
  <h1>
    Under Construction
    <br />
    Coming Soon
  </h1>
</div>
`;
